README.tfdoc.hcl

```
@@ -0,0 +1,826 @@
+        default     = false
+        description = <<-END
+          Specifies whether to enable Amazon ECS managed tags for the tasks
+          within the service.
+        END
+      }
+
+      variable "enable_execute_command" {
+        type        = bool
+        default     = false
+        description = <<-END
+          Specifies whether to enable Amazon ECS Exec for the tasks within the
+          service.
+        END
+      }
+
+      variable "force_new_deployment" {
+        type        = bool
+        default     = false
+        description = <<-END
+          Enable to force a new task deployment of the service. This can be used
+          to update tasks to use a newer Docker image with same image/tag
+          combination (e.g. `myimage:latest`), roll Fargate tasks onto a newer
+          platform version, or immediately deploy `ordered_placement_strategy`
+          and `placement_constraints updates`.
+        END
+      }
+
+      variable "wait_for_steady_state" {
+        type        = bool
+        default     = false
+        description = <<-END
+          If `true`, Terraform will wait for the service to reach a steady
+          state—as in, the ECS tasks you wanted are actually deployed—before
+          `apply` is considered complete.
+        END
+      }
+
+      variable "ordered_placement_strategy" {
+        type        = list(ordered_placement_strategy)
+        default     = []
+        description = <<-END
+          Service level strategy rules that are taken into consideration during
+          task placement. List from top to bottom in order of precedence.
+          Updates to this configuration will take effect next task deployment
+          unless `force_new_deployment` is enabled. The maximum number of
+          `ordered_placement_strategy` blocks is `5`. For details please
+          see [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_service#ordered_placement_strategy).
+        END
+
+        attribute "type" {
+          required    = true
+          type        = string
+          description = <<-END
+            Type of placement strategy. Must be one of: `binpack`, `random`, or
+            `spread`.
+          END
+        }
+
+        attribute "field" {
+          type        = string
+          description = <<-END
+            For the `spread` placement strategy, valid values are `instanceId` (or
+            `host`, which has the same effect), or any platform or custom
+            attribute that is applied to a container instance. For the `binpack`
+            type, valid values are `memory` and `cpu`. For the `random` type, this
+            attribute is not needed. For more information, see
+            [Placement Strategy](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PlacementStrategy.html).
+          END
+        }
+      }
+
+      variable "placement_constraints" {
+        type        = set(placement_constraint)
```

```suggestion
        type        = list(placement_constraint)
```

this is a list afaik

---

Same with this:
```
"placement_constraints": {
                                "nesting_mode": "set",
                                "block": {
```